openedx/core/djangoapps/user_api/models.py

```diff
@@ -282,3 +316,14 @@ def get_retirement_for_retirement_action(cls, username):
 
     def __unicode__(self):
         return u'User: {} State: {} Last Updated: {}'.format(self.user.id, self.current_state, self.modified)
+
+
+@receiver(models.signals.post_delete, sender=UserRetirementStatus)
+def remove_pending_retirement_request(sender, instance, **kwargs):   # pylint: disable=unused-argument
```

Comment:
I'm not sure we've hashed out a CS procedure for revoking a retirement request, or if that's even (easily) possible given that we immediately hash out the username, etc. We should really nail that down, because the ABORTED status was intended as a holder for people who took this action, and I think in the current state of this PR a learner in that state, but still with a UserRetirementRequest row, would still be subject to all of the restrictions put on a retiring learner.

Best thing might be to get rid of the ABORTED state and just make retirement cancellation delete the UserRetirementState row instead.

Reply:
I'm fine with removing the ABORTED state, can't think of any other reason to have it.

Taking this cancellation workflow one step further, if CS deletes the state row and this handler gets triggered, the user is left in an unrecoverable state since the original_email was erased forever.  Just spoke with John in person and we agreed that a disco ticket regarding cancellation workflow is warranted, if it doesn't already exist.